### DISCUSSION TITLE
Notes held before recording starts aren't captured in recording

### DISCUSSION THREAD

**Discussion by trappar** [SynthstromAudible/DelugeFirmware#257] [Ideas]
When using the keyboard or drum keyboard, a pad that's being held before recording starts will not be included in the recording. This is especially relevant when playing in mono/legato patterns where you might hold a low note while alternating over higher notes.

To replicate with the synth keyboard:
1. Arm recording
2. Hold down a note
3. Press play
4. Let go of the note before the clip finishes
5. Observe that no note exists in the recorded clip

It's basically the same with the drum keyboard, but in that case it probably only makes sense to try to catch sounds which are using the cut, loop, or stretch modes.

**Comment by weavermedia** [comment]
If anything was implemented to capture existing held notes when recording starts I hope it would be optional because the issue you described sounds like intended behavior to me.

I can see the desire to capture existing held notes but it's unusual enough behavior that I think it might surprise most people.

**Comment by litui** [comment]
I would agree with @weavermedia. While this behaviour is intuitive as audio-recording behaviour, it's not something I've seen in the midi world previously. I can't even imagine how it could be done particularly well.

**Comment by weavermedia** [comment]
Can this be closed as 'intended behavior'? Then a feature request for this 'pre-record MIDI' behavior could be posted to the forum and/or as a working draft PR here.

**Comment by sichtbeton** [comment]
agreed, its as expected in MIDI

**Comment by trappar** [comment]
Did a bit of an audit to see what other devices/software do when it comes to this kind of behavior. Here's my results:

Note held before recording started, then start recording.

| Device/Software| Is note recorded? | Extra details |
|--------|--------|--------|
| Ableton | Yes | As a long time Ableton user, I think this is where my expectation that it would work this way came from. |
| Bitwig | No |  |
| Akai MPC One | No | |
| NI Maschine+ | No | |
| Roland MC-101 | Yes | |
| TE OP-1 | Yes | Only because of the nature of recording audio |
| TE OP-Z | N/A | OP-Z handles this very gracefully and won't allow a situation that could be confusing to users. You either hold rec + play and press a key, which then starts recording with that key in the first step, or you hold a key and press rec + play + play and it disables the key when starting transport. |
| Dirtywave M8 | Yes(ish) | M8 records any incoming midi on channel 9 (by default) to the current channel. The issue is that it does this even when stopped. This is closer to the yes behavior than no, but doesn't really qualify. |

So I'd put this squarely in the "nice to have" category based on its support elsewhere.
